### PR TITLE
[Feature] Add/Remove reducer with HashEngine

### PR DESCRIPTION
**Pull request by kittinunf**
### Description

We make the HashEngine support both adding/removing Reducer dynamically.

### CODE REVIEW

File: cored/src/commonMain/kotlin/com/github/kittinunf/cored/store/ReduxStore.kt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package com.github.kittinunf.cored.store
 
+import com.github.kittinunf.cored.AnyReducer
```
Comment (zalewskise) [inline cored/src/commonMain/kotlin/com/github/kittinunf/cored/store/ReduxStore.kt]: Do we need it?

File: cored/src/commonMain/kotlin/com/github/kittinunf/cored/engine/HashEngine.kt [hunk 2]
```diff
@@ -51,4 +55,13 @@ internal class HashEngine<S : Any, A : Any>(
...
     }
+
+    override fun addReducer(actionReducer: ActionReducer<S, Any>) {
+        val (key, reducer) = actionReducer
+        reducerMap.put(key, reducer)
+    }
+
+    override fun removeReducer(key: Any): Boolean {
+        return reducerMap.remove(key) != null
+    }
```
Comment (zalewskise) [inline cored/src/commonMain/kotlin/com/github/kittinunf/cored/engine/HashEngine.kt]: I would suggest to have method with same parameter type as for `addReducer()`:
```
override fun removeReducer(actionReducer: ActionReducer<S, Any>): Boolean { ... }
```
Assuming user of the API might not know at first look what's the `key` (as he is not passing the explicit `key` when adding reducer) and pass `ActionReducer` (as the type is `Any`, it's possible) to this method, which will not work.

Comment (zalewskise) [reply cored/src/commonMain/kotlin/com/github/kittinunf/cored/engine/HashEngine.kt]: Same like for middleware :)

Comment (kittinunf) [reply cored/src/commonMain/kotlin/com/github/kittinunf/cored/engine/HashEngine.kt]: kk!

**Review by zalewskise — APPROVED** [review]
🐐 🥇 !

**Pull request merged**